### DISCUSSION TITLE
Support Nuxt.js framework

### DISCUSSION THREAD

**Discussion by septatrix** [nextauthjs/next-auth#6825] [Ideas]
### Description 📓

Nuxt.js 3 currently has no great authentication solution. The solution used for v2 is currently lacking maintainers and has not seen a lot of activity in the recent months so v3 support is still far on the horizon. Even then it supports far fewer providers than this project. It would be great if this project could provide support for Nuxt.js and fill this void.

`@auth/core` already supports Next.js, SvelteKit and SolidStart so adding support for Nuxt.js would fit will into the scope and roadmap of this project with the scope to provide an auth solution for all frameworks.

### How to reproduce ☕️

Go to <https://authjs.dev/reference/core> and see that there is no `@auth/nuxtjs` reference

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by septatrix** [comment]
I found the [nuxt playground app](https://github.com/nextauthjs/next-auth/tree/main/apps/playgrounds/nuxt) in this repo yesterday and so far it seems to work fine. I will experiment a bit more but so far it looks well enough that one might consider promoting it to a standalone package.

**Comment by Hebilicious** [comment]
Is the official integration being worked on already behind the scenes ?
I know https://github.com/sidebase/nuxt-auth/ is based on next-auth and is relatively popular (484 stars atm), but I'm not sure both teams are in touch ?
I'm happy to open a PR here if that's needed. What's currently in the playground is roughly equivalent to the svelte-kit integration, agreed it could be promoted to a package.

**Comment by Hebilicious** [comment]
I started working on a Nuxt module : https://github.com/Hebilicious/authjs-nuxt 
This is edge-compatible and should support all Auth.js feature. I can open a PR so this can be merged here. @balazsorban44 how should we proceed ?

**Reply by toniengelhardt** [reply]
Have you been in touch with Nuxt authors on this @Hebilicious? 

I inquired several times about the state of auth with Nuxt, but never got a proper answer on what the plan is. `nuxt-auth` seems dead, `nuxt-alt/auth` works kind of, but it's buggy. Would love to eventually see a solution that can properly authenticate a non-SSR Nuxt app with e.g. a Django backend, social auth and such.

Thanks for your efforts 🙏🏽

**Reply by Hebilicious** [reply]
@toniengelhardt This is an auth.js related discussions, so not sure this is the right place to discuss this.

If you are using Nuxt as the frontend framework with something like Django/Laravel, you should wire the auth yourself. Ideally users of these frameworks will contribute a dedicated nuxt module.

An official nuxt module that handles all use cases (similar to the nuxt2 one) is a massive undertaking and would take a significant amount of work, and I don't think anyone is working on such a thing at this time. It would be great if a company decided to fund such an endeavour, but in the meantime you'll have to rely existing OSS solutions or develop your own. Note that you do not need a module to authenticate with a nuxt app, they do make it more convenient for you, but they are not necessary.

**Reply by toniengelhardt** [reply]
Oh sorry, I didn't realize Auth.js only handles the backend logic, I thought it was a complete auth solution 🙈

I understand that it is a massive undertaking, but it also doesn't make much sense if everyone has to re-invent the wheel from scratch. Hacking your own solutions also opens the door for security risks if you're not an expert IMO.

**Comment by Syed-Sheharyar** [comment]
What's the current state of this discussion? Does Nuxt JS further proceed?

**Comment by fir3bird** [comment]
I am also very interested in the current state. I can see from the docs that this is "planned" but unfortunately without an eta or similar.
I also wonder why the Nuxt Auth project from @Hebilicious (https://github.com/Hebilicious/authjs-nuxt) is not used as the basis for further development. I think this is already a good starting point, don't you?
